docs/spark/how-to-guides/batch-processing.md

```diff
@@ -0,0 +1,124 @@
+    }
+   ```
+
+1. Inside the `Run` function, add the following code to establish a new SparkSession. The SparkSession is the entry point to programming Spark with the Dataset and DataFrame API. By calling the `spark` object, you can access Spark and DataFrame functionality throughout your program.
+
+   ```csharp
+   SparkSession spark = SparkSession
+                .Builder()
+                .AppName("GitHub and Spark Batch")
+                .GetOrCreate();
+   ```
+
+1. Read the input file into a DataFrame, which is a distributed collection of data organized into named columns.
+
+   ```csharp
+   DataFrame projectsDf = spark
+       .Read()
+       .Schema("id INT, url STRING, owner_id INT, " +
+       "name STRING, descriptor STRING, language STRING, " +
+       "created_at STRING, forked_from INT, deleted STRING" +
+       "updated_at STRING")
+       .Csv(args[0]);
+
+   projectsDf.Show();
+   ```
+
+1. Use `DataFrame.Na()` to drop rows with NA (null) values, and the `DataFrameNaFunctions.Drop()` method to remove certain columns from your data. This helps prevent errors if you try to analyze null data or columns that are not relevant to your final analysis.
+
+   ```csharp
+   // Drop any rows with NA values
+   DataFrameNaFunctions dropEmptyProjects = projectsDf.Na();
+   DataFrame cleanedProjects = dropEmptyProjects.Drop("any");   
+   // Remove unnecessary columns
+   cleanedProjects = cleanedProjects.Drop("id", "url", "owner_id");
+   cleanedProjects.Show();
+   ```
+
+1. Spark SQL allows you to make SQL calls on your data. It's common to combine user-defined functions and Spark SQL so to apply a user-defined function to all rows of your DataFrame.
+
+   You can specifically call `spark.Sql` to mimic standard SQL calls seen in other types of apps. You can also call methods like `GroupBy` and `Agg` to specifically combine, filter, and perform calculations on your data.
+
+   ```csharp
+   // Average number of times each language has been forked
+   DataFrame groupedDF = cleanedProjects
+       .GroupBy("language")
+       .Agg(Avg(cleanedProjects["forked_from"])   
+   // Sort by most forked languages first
+   groupedDF.OrderBy(Desc("avg(forked_from)")).Show(   
+   spark.Udf().Register<string, bool>(
+       "MyUDF",
+       (date) => DateTime.TryParse(date, out DateTi   convertedDate) &&
+           (convertedDate > s_referenceDate)   
+   cleanedProjects.CreateOrReplaceTempView("dateView"   
+   DataFrame dateDf = spark.Sql(
+       "SELECT *, MyUDF(dateView.updated_at) AS datebefo   FROM dateView");
+   dateDf.Show();
+   ```
+
+1. Call `spark.Stop()` to end the SparkSession.
+
+1. Use `spark-submit` to run your app.
+
+   ```console
+   spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local C:\GitHub\spark\src\scala\microsoft-spark-2.4.x\target\microsoft-spark-2.4.x-0.5.0.jar Microsoft.Spark.CSharp.Examples.exe Batch.GitHubProjects %SPARK_HOME%\examples\src\main\resources\projects_smaller.csv
```

Comment:
```suggestion
   spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local /path/to/microsoft-spark-<version>.jar Microsoft.Spark.CSharp.Examples.exe Sql.Batch.GitHubProjects /path/to/projects_smaller.csv
```

Reply:
We used this `spark-submit` command example in the final readme (so that it isn't version specific/needs to be updated with each new version number that comes out).